crates/subspace-core-primitives/src/objects.rs

```diff
@@ -134,10 +134,13 @@ pub enum GlobalObject {
     #[codec(index = 0)]
     #[cfg_attr(feature = "serde", serde(rename_all = "camelCase"))]
     V0 {
-        /// Piece index where object is contained (at least its beginning, might not fit fully)
+        /// Piece index where object is contained (at least its beginning, might not fit fully).
+        /// The index and offset must be first, so that the Ord derives are valid.
```

Comment:
I'm not convinced it is a good thing to have index first.

Ordering is not invalid either way, but thinking about actual use cases I thought that storing these in a sorted vector in memory on on disk would be a fairly reasonable one and in that case it is actually beneficial to make hash the first field in the data structure such that it is possible to quickly find things by hash using binary search. This is significantly more efficient both from memory usage point of view and from performance standpoint.

On a similar note enum variant should probably move to `GlobalObjectMapping` data structure otherwise we have a lot of redundancy due to data alignment and generally waste one byte per entry for no reason when returning a bunch of elements.